python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -55,20 +59,140 @@ def start(self, initialization_hook: Optional[Callable[[], None]] = None):
             self.worker_group.execute(initialization_hook)
         self._backend.on_start(self.worker_group, self._backend_config)
 
-    def run(self, train_func: Callable[[], T]) -> List[T]:
-        """Executes a training function on all workers.
+    def start_training(self, train_func: Callable[[], T]) -> None:
+        """Executes a training function on all workers in a separate thread.
 
         Args:
             train_func (Callable): The training function to run on each worker.
+        """
+
+        # First initialize the session.
+        def initialize_session(world_rank):
+            thread = PropagatingThread(target=train_func, daemon=True)
+            try:
+                init_session(training_thread=thread, world_rank=world_rank)
```

Comment:
I'm wondering if we can move all of the references to `Thread`s into the `Session` class so that `Backend` doesn't need to directly interact with threads.

- `initialize_session`: This one I'm a bit iffy on whether we want to initialize the thread here and pass it in to `init_session`, or just have `init_session` take in a `target`.
- `train_async`: Can we create a simple `Session.start` method that calls `self.training_thread.start()`?
- `end_training`: Similarly, we can wrap `training_thread.join()` in a method called `finish()` or `end()`.
- `get_next`: This one has a little bit more logic but seems like it can exist in `Session` as well?

Reply:
Changed.